apps/skilavottord/web/screens/AccessControl/components/AccessControlUpdate/AccessControlUpdate.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { FC, useEffect } from 'react'
+import { useForm } from 'react-hook-form'
+
+import { Option } from '@island.is/island-ui/core'
+import { ModalProps } from '@island.is/skilavottord-web/components'
+import {
+  AccessControl,
+  UpdateAccessControlInput,
+} from '@island.is/skilavottord-web/graphql/schema'
+
+import { AccessControlModal } from '../AccessControlModal/AccessControlModal'
+
+interface AccessControlUpdateProps
+  extends Omit<
+    ModalProps,
+    'onContinue' | 'continueButtonText' | 'cancelButtonText'
+  > {
+  onSubmit: (partner: UpdateAccessControlInput) => Promise<void>
+  recyclingPartners: Option[]
+  roles: Option[]
+  currentPartner: AccessControl
+}
+
+export const AccessControlUpdate: FC<AccessControlUpdateProps> = ({
+  title,
+  text,
+  show,
+  onCancel,
+  onSubmit,
+  recyclingPartners,
+  roles,
+  currentPartner,
+}) => {
+  const { control, errors, setValue, handleSubmit } = useForm({
+    mode: 'onChange',
+  })
+
+  useEffect(() => {
```

Comment:
What's going on here in this effect?

Reply:
Setup fields of the partner being edited, I was able to simplify it though.